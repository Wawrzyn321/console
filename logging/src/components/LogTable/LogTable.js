import React from 'react';
import PropTypes from 'prop-types';
import './LogTable.scss';

LogTable.propTypes = {
  entries: PropTypes.arrayOf(PropTypes.object.isRequired).isRequired,
};

export default function LogTable({ entries }) {
  function renderEntries() {
    return entries.map(entry => (
      <tr key={entry.timestamp}>
        <td className="caption-muted">{entry.timestamp}</td>
        <td className="caption-muted">{entry.log}</td>
      </tr>
    ));
  }

  return (
    <table className="fd-table fd-has-margin-regular">
      <thead>
        <tr>
          <th colSpan="2" className="log-table__pre-header fd-has-type-0">
            Stream Name
          </th>
        </tr>
        <tr>
          <th className="caption-muted">Timestamp</th>
          <th className="caption-muted">Log</th>
        </tr>
      </thead>
      <tbody>
        {!!entries.length ? (
          renderEntries()
        ) : (
          <tr>
            <td colSpan="2" className="log-table__no-entries-text">
              No entries
            </td>
          </tr>
        )}
      </tbody>
    </table>
  );
}
