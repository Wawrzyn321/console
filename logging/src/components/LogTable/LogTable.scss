.aaa {
  background: red !important;
  &::before {
    content: ${props => (props.noIcon ? "''" : "'\uE00D'")};
    position: absolute;
    font-family: SAP-icons;
    font-size: 16px;
    font-style: normal;
    pointer-events: none;
    right: 10px;
    top: 9px;
  }
}
