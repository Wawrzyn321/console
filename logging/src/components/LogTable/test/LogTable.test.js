import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import LogTable from './../LogTable';

describe('LogTable', () => {
  it('Renders with minimal props', () => {
    const component = renderer.create(
      <LogTable
        entries={[
          { timestamp: 'a', log: 'b' },
          { timestamp: 'c', log: 'd' },
          { timestamp: 'e', log: 'f' },
        ]}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Displays "No entries" button when there are no entries', () => {
    const component = shallow(<LogTable entries={[]} />);
    expect(component.exists('.log-table__no-entries-text')).toBeTruthy();
  });
});
