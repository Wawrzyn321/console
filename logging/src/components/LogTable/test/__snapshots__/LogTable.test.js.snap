// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogTable Renders with minimal props 1`] = `
<table
  className="fd-table fd-has-margin-regular"
>
  <thead>
    <tr>
      <th
        className="log-table__pre-header fd-has-type-0"
        colSpan="2"
      >
        Stream Name
      </th>
    </tr>
    <tr>
      <th
        className="caption-muted"
      >
        Timestamp
      </th>
      <th
        className="caption-muted"
      >
        Log
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="caption-muted"
      >
        a
      </td>
      <td
        className="caption-muted"
      >
        b
      </td>
    </tr>
    <tr>
      <td
        className="caption-muted"
      >
        c
      </td>
      <td
        className="caption-muted"
      >
        d
      </td>
    </tr>
    <tr>
      <td
        className="caption-muted"
      >
        e
      </td>
      <td
        className="caption-muted"
      >
        f
      </td>
    </tr>
  </tbody>
</table>
`;
